src/analysis/fourier/add_circle.lean

```diff
@@ -266,64 +286,114 @@ end
 
 end monomials
 
-section fourier
+section scope_hT -- everything from here on needs `0 < T`
 variables [hT : fact (0 < T)]
 include hT
 
-/-- We define `fourier_series` to be a `ℤ`-indexed Hilbert basis for `Lp ℂ 2 haar_add_circle`,
+
+section fourier_coeff
+
+/-- The Fourier coefficients of a function `add_circle T → ℂ`, defined as the integral over
+`add_circle T` of `λ t, fourier (-n) t * f t`. -/
+def fourier_coeff (f : add_circle T → ℂ) (n : ℤ) : ℂ :=
+∫ (t : add_circle T), fourier (-n) t * f t ∂ haar_add_circle
+
+/-- The Fourier coefficients of an `ae_strongly_measurable` function can be computed as an integral
+over `[a, a + T]` for any real `a`. -/
+lemma fourier_coeff_eq_interval_integral
+  {f : add_circle T → ℂ} (hf : ae_strongly_measurable f volume) (n : ℤ) (a : ℝ) :
```

Comment:
This lemma shouldn't need `hf`. See `set_integral_eq` in `measure_theory/group/fundamental_domain`.

Reply:
I agree that `hf` shouldn't be needed, but the dependence isn't my creation; it's inherited straight from the existing code for `add_circle.interval_integral_preimage`.

Reply:
It definitely does not seem to be straightforward to re-prove `add_circle.interval_integral_preimage` to remove the ae_strongly_measurable hypothesis.

Reply:
It wasn't too bad in the end: I now have a proof of `∫ a in Ioc t (t + T), f a = ∫ b : add_circle T, f b` without any a priori measurability assumptions, as a consequence of defining a `measurable_equiv` between the circle and `Ioc t (t + T)`. I'll post it as a separate PR in a moment.

Reply:
Great, thanks a lot for looking into this!